golemfactory/yagna
# First working zksync transfer from yagna

Adds transfer to the `payment-driver/zksync`

When calling SchedulePayment on the zksync driver:
- Generate zksync private key -> wallet from ethereum signed message
- Create zksync transaction + ethereum signed message
- Send transaction to zksync provider

This does not include persistence, payments are done immediately.

To test:
- Generate `{provider,requestor}.key.json` files running the javascript basic scenario with `-s` parameter.
- Run payment_api example ( on a **clean** database )loading the keys and zksync driver
```
cd core/payment
cargo run --example payment_api -- --driver zksync --provider-key-path ./provider.key.json --requestor-key-path ./requestor.key.json
```
- Run invoice flow:
```
cd core/payment
cargo run --example invoice_flow
```
- check http://rinkeby.zkscan.io to see the transaction
